iceoryx_posh/source/roudi/service_registry.cpp

```diff
@@ -119,7 +128,7 @@ void ServiceRegistry::find(ServiceDescriptionVector_t& searchResult,
 
             if (match)
             {
-                searchResult.emplace_back(*entry);
+                callable(*entry);
```

Comment:
Oh I was so strictly against making the `cxx::function_ref` nullable since such simple mistakes like this can happen. This is so dangerous!

But you have to check the `callable` first before calling it since when it does not contain a callable this produces either undefined behavior or a call to the error handler - or a terminate.

So please add:
```
if ( !callable ) {
  return;
}
```

at the top of the method.

Reply:
@FerdinandSpitzschnueffler Please also add a test case for that, where you call `find()` with supplying a `nullptr` as last argument.

Reply:
@mossmaurice I already added a test with an empty function_ref. Calling `find()` with a nullptr would just check that an empty function_ref is created from a nullptr and that should be checked in the function_ref tests.